core/arch/arm32/kernel/tee_ta_manager.c

```diff
@@ -1060,11 +1060,33 @@ static void tee_ta_destroy_context(struct tee_ta_ctx *ctx)
 	free(ctx);
 }
 
+/* check if requester (client ID) matches session initial client */
+static TEE_Result check_client(struct tee_ta_session *s, const TEE_Identity *id)
+{
+	if (id == KERN_IDENTITY)
+		return TEE_SUCCESS;
+
+	if (id == NULL) {
+		if (s->clnt_id.login == TEE_LOGIN_TRUSTED_APP) {
+			DMSG("nsec attempts TA session usurpation");
```

Comment:
It looks like someone used a dictionary. Wouldn't "nsec tries to hijack TA session" be easier to understand?